src/ListSource.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_list_pages;
+
+use Drupal\Core\Plugin\PluginBase;
+use Drupal\search_api\Datasource\DatasourceInterface;
+
+/**
+ * List source object.
```

Comment:
What is a list source object? Let's have some info here. 

Also, let's have an interface.